feat(applications-dynamics-ax-cma): initial release CTOR-2354 (#5558)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-dynamics-ax-cma.md

---
id: applications-dynamics-ax-cma
title: Dynamics AX CMA
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import CMAprerequisites from './_cma-prerequisites.mdx';

The connector allows you to monitor:
* Windows Server OS from 2003 SP2 version

> * Windows workstations from the XP version

Reviewer: ```suggestion
* Windows workstations from the XP version.
```

## Pack assets

### Templates

The Monitoring Connector **Dynamics AX CMA** brings a host template:

* **App-Dynamics-AX-CMA-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="App-Dynamics-AX-CMA-custom" label="App-Dynamics-AX-CMA-custom">

| Service Alias               | Service Template                                       | Service Description                                       |
|:----------------------------|:-------------------------------------------------------|:----------------------------------------------------------|

> | RIS-Import-Input            | App-Dynamics-AX-RIS-Import-Input-CMA-custom            | Check import files presence                               | 

Reviewer: ```suggestion
| RIS-Import-Input            | App-Dynamics-AX-RIS-Import-Input-CMA-custom            | Check whether files must be imported                               | 
```

The installation procedures for monitoring connectors are slightly different depending on [whether your license is offline or online](../getting-started/how-to-guides/connectors-licenses.md).

1. If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the connector displayed within the
**Configuration > Monitoring Connectors Manager** menu.
If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-applications-dynamics-ax-cma
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-applications-dynamics-ax-cma
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
apt install centreon-pack-applications-dynamics-ax-cma
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-applications-dynamics-ax-cma
```

</TabItem>
</Tabs>

> 2. Whatever the license type (*online* or *offline*), install the **Windows NSClient API** connector through

Reviewer: ```suggestion
2. Whatever the license type (*online* or *offline*), install the **Dynamics AX CMA** connector through
```

### Filter expressions

> Applies for FILTER, WARNINGSTATUS, CRITICALSTATUS.

Filter syntax is similar to C/SQL:

- Numeric operators: `==`, `!=`, `>`, `<`, `>=`, `<=`
- Logical: `&&` (AND), `||` (OR)
- String equality: `==`, `!=` (note: single `=` is **not** valid)
- IN/NOT IN: `filename in ('myfile.txt')`, `version in ('1.0', '1.1')`

Supported file metadata labels:

- size           (in bytes, supports units: K, Ko, M, Mo, G, Go)
- line_count     (line counting in txt file)
- creation       (file age in seconds since creation, supports units : w, d, h, m, s) no unit = s. Unit can’t be composed (ie : 1d3h)
- access         (file age in seconds since last access, supports units : w, d, h, m, s) no unit = s. Unit can’t be composed (ie : 1d3h)
- written        (file age in seconds since last modification, supports units : w, d, h, m, s) no unit = s. Unit can’t be composed (ie : 1d3h)

> - filename       (name of the file, string comparison)

Reviewer: same as FR and the rest of the file

- "size > 200m && extension == '.dll'"    # Large DLLs
- "count &lt;= 0"                            # No file found
- "filename in ('myfile.txt')"            # Specific file by name
- "filename == 'myfile.txt'"              # Specific file by name (alternative)

#### File age check

File age can be checked using 3 metadata labels, which can be mixed using logical operators :

- creation       (file age in seconds since creation, supports units : w, d, h, m, s) no unit = s. Unit can’t be composed (ie : 1d3h)
- access         (file age in seconds since last access, supports units : w, d, h, m, s) no unit = s. Unit can’t be composed (ie : 1d3h)
- written        (file age in seconds since last modification, supports units : w, d, h, m, s) no unit = s. Unit can’t be composed (ie : 1d3h)

_“I want to trigger a CRITICAL alert if at least one file of my test directory has not be updated since 1 day or more, and WARNING alert if more than 12 hours and less than 1 day”_

```
PATH= C:/Users/User/Documents/test
PATTERN= *.*
MAXDEPTH= -1,
DETAILSYNTAX= {filename}: {size}
WARNINGSTATUS= written > 12h
CRITICALSTATUS= written > 1d
WARNING= 0
CRITICAL= 0
```

#### File size check

> _“I want to trigger CRITICAL alert if at least 1 DLLs in System32 (including subdirs without recursivity) size is >100M, and trigger WARNING alert if at least 2 DLLs size is >10M”_

Reviewer: ```suggestion
_"I want to trigger CRITICAL alert if at least 1 DLL in System32 (including subdirs without recursivity) size is >100M, and trigger WARNING alert if at least 2 DLLs size is >10M"_
```

The extension filter can be done using PATTERN or FILTER.

```
PATH= C:/Windows/System32
PATTERN= *.dll
MAXDEPTH= 1,
OUTPUTSYNTAX= {status}: {problem_count}/{count} DLLs have issues: {problem_list}
DETAILSYNTAX= {filename}: {size} {version}
FILTER= extension == '.dll'
WARNINGSTATUS= size > 10m
CRITICALSTATUS= size > 100m
WARNING= 1
CRITICAL= 0
```

Note:

- If "line_count" is used in any filter or output, line count calculation will be enabled (may impact performance).
- Paths, patterns, and filters are case-insensitive on Windows.
- Use `/` as the path separator instead of `\` (e.g., `C:/Users/...`). Backslashes require extra escaping and may cause errors.

#### File presence check

> _“I want to trigger a CRITICAL alert if file is not present“_

Reviewer: ```suggestion
_"I want to trigger a CRITICAL alert if a specific file is not present"_
```

Reviewer: ?

```
PATH= C:/Users/User/Documents/test
PATTERN= myfile.txt
MAXDEPTH= -1,
DETAILSYNTAX= {filename}: {size}
WARNINGSTATUS=
CRITICALSTATUS= count < 0
WARNING=
CRITICAL= 0
```

> _“I want to trigger a CRITICAL alert if at least one file is present“_

Reviewer: ```suggestion
_"I want to trigger a CRITICAL alert if at least one file is present"_
```

Files matching the WARNINGSTATUS/CRITICALSTATUS filters are considered WARNING/CRITICAL.
You can combine with WARNING/CRITICAL to require multiple matches before changing the global state.

### Examples

#### Filters

> Applies for FILTER, WARNINGSTATUS, CRITICALSTATUS.

- "size > 50M"                            # File larger than 50 MB
- "extension == '.bak'"                   # Backup files
- "size > 200m && extension == '.dll'"    # Large DLLs
- "count &lt;= 0"                            # No file found
- "filename in ('myfile.txt')"            # Specific file by name
- "filename == 'myfile.txt'"              # Specific file by name (alternative)

#### File age check

File age can be checked using 3 metadata labels, which can be mixed using logical operators :

- creation       (file age in seconds since creation, supports units : w, d, h, m, s) no unit = s. Unit can’t be composed (ie : 1d3h)
- access         (file age in seconds since last access, supports units : w, d, h, m, s) no unit = s. Unit can’t be composed (ie : 1d3h)
- written        (file age in seconds since last modification, supports units : w, d, h, m, s) no unit = s. Unit can’t be composed (ie : 1d3h)

> _“I want to trigger a CRITICAL alert if at least one file of my test directory has not be updated since 1 day or more, and WARNING alert if more than 12 hours and less than 1 day”_

Reviewer: ```suggestion
_"I want to trigger a CRITICAL alert if at least one file of my test directory has not be updated since 1 day or more, and WARNING alert if more than 12 hours and less than 1 day"_
```

```
PATH= C:/Users/User/Documents/test
PATTERN= *.*
MAXDEPTH= -1,
DETAILSYNTAX= {filename}: {size}
WARNINGSTATUS= written > 12h
CRITICALSTATUS= written > 1d
WARNING= 0
CRITICAL= 0
```

#### File size check

> _“I want to trigger CRITICAL alert if at least 1 DLLs in System32 (including subdirs without recursivity) size is >100M, and trigger WARNING alert if at least 2 DLLs size is >10M”_

Reviewer: ```suggestion
_"I want to trigger a CRITICAL alert if at least 1 DLL in System32 (including subdirs without recursivity) size is >100M, and trigger a WARNING alert if at least 2 DLLs' size is >10M”_
```

The extension filter can be done using PATTERN or FILTER.

```
PATH= C:/Windows/System32
PATTERN= *.dll
MAXDEPTH= 1,
OUTPUTSYNTAX= {status}: {problem_count}/{count} DLLs have issues: {problem_list}
DETAILSYNTAX= {filename}: {size} {version}
FILTER= extension == '.dll'
WARNINGSTATUS= size > 10m
CRITICALSTATUS= size > 100m
WARNING= 1
CRITICAL= 0
```

Note:

- If "line_count" is used in any filter or output, line count calculation will be enabled (may impact performance).
- Paths, patterns, and filters are case-insensitive on Windows.
- Use `/` as the path separator instead of `\` (e.g., `C:/Users/...`). Backslashes require extra escaping and may cause errors.

#### File presence check

> _“I want to trigger a CRITICAL alert if file is not present“_

Reviewer: ```suggestion
_"I want to trigger a CRITICAL alert if a specific file is not present"_
```

Reviewer: or if no file is present?

```
PATH= C:/Users/User/Documents/test
PATTERN= myfile.txt
MAXDEPTH= -1,
DETAILSYNTAX= {filename}: {size}
WARNINGSTATUS=
CRITICALSTATUS= count < 0
WARNING=
CRITICAL= 0
```

> _“I want to trigger a CRITICAL alert if at least one file is present“_

Reviewer: ```suggestion
_"I want to trigger a CRITICAL alert if at least one file is present"_
```
